### PR TITLE
Move winston NodeJS transports to cli package

### LINKED ISSUE #2381: Not able to turn on debug for a specific module
**Describe the bug**

By default we have `info` debug log level for all modules. If I use "`--logger.chain.level debug`" then I see debugs from all modules:

```
e5a33456475d51e6fc79a652d
Apr-15 05:02:24.898 [CHAIN]         verbose: Clock slot slot=162312
Apr-15 05:02:24.899 [CHAIN]           debug: Block pools:  pendingBlocks=0, currentSlot=162312
Apr-15 05:02:25.825 [CHAIN]         verbose: Block processed slot=154815, root=0xbfc9896b853ef6063de3c206071e6442cc44dde6966451f4f1ec2ca2e2a33cb2
Apr-15 05:02:26.045 [CHAIN]         verbose: New chain head headSlot=154815, headRoot=0xbfc9896b853ef6063de3c206071e6442cc44dde6966451f4f1ec2ca2e2a33cb2
Apr-15 05:02:26.046 [NETWORK]         debug: Req  dialing peer method=ping, encoding=ssz_snappy, agentVersion=N/A, peer=16...LibfSe, requestId=1815
Apr-15 05:02:26.047 [NETWORK]         debug: Req  dialing peer method=ping, encoding=ssz_snappy, agentVersion=nimbus, peer=16...PpWAqp, requestId=1816
Apr-15 05:02:26.047 [NETWORK]         debug: Req  dialing peer method=ping, encoding=ssz_snappy, agentVersion=N/A, peer=16...gAj37q, requestId=1817
Apr-15 05:02:26.048 [NETWORK]         debug: Req  dialing peer method=status, encoding=ssz_snappy, agentVersion=nimbus, peer=16...L1vsyn, requestId=1818
Apr-15 05:02:26.371 [NETWORK]         debug: Req  sending request method=ping, encoding=ssz_snappy, agentVersion=N/A, peer=16...LibfSe, requestId=1815, requestBody=0
Apr-15 05:02:26.373 [NETWORK]         debug: Req  sending request method=status, encoding=ssz_snappy, agentVersion=nimbus, peer=16...L1vsyn, requestId=1818, requestB
```
so it's like "all or nothing"

**Expected behavior**

We should be able to turn on "debug" level for a specific module.

**Steps to Reproduce**
+ Sync Prater with `--logger.chain.level debug`

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

`@lodestar/utils` package is meant to contain misc utilities re-used across multiple packages that is **platform agnostic**, and compatible in the browser.

Defining a common logger for all packages makes sense, **but** including a NodeJS only transport (FileRotate) breaks the above invariant.

Flags of type `--logger.chain.level debug` have been broken for a long time. This is due to the way winston handles log level at the transport level and doesn't allow customization on child logger.

**Description**

- Move winston NodeJS transports to cli package
- Keep generic platform agnostic logger code in utils
- Re-enable per-module log level setting

Also, testing this change found out why file rotate does not clean old files. The library https://www.npmjs.com/package/winston-daily-rotate-file can only delete files that it know of if they are present in the audit file. The audit file location is by default the process cwd, which is not persisted if dockerized. Forcing the audit file to be in the same directory as the log file fixes the issue

https://github.com/ChainSafe/lodestar/blob/d6a7074c0de42b6664821d09b6e41846ae80d77f/packages/cli/src/util/loggerTransports.ts#L38

closes https://github.com/ChainSafe/lodestar/issues/2381

### CODE REVIEW

File: packages/beacon-node/test/e2e/chain/lightclient.test.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ describe("chain / lightclient", function () {
       });
     }).then(async (head) => {
       // Initialize lightclient
-      loggerLC.important("Initializing lightclient", {slot: head.slot});
+      loggerLC.info("Initializing lightclient", {slot: head.slot});
```
Comment (dapplion) [inline packages/beacon-node/test/e2e/chain/lightclient.test.ts:116]: .important API is not used anymore in production and forces creators of custom loggers to add it

File: packages/beacon-node/test/e2e/db/api/beacon/repositories/blockArchive.test.ts [hunk 2]
```diff
@@ -70,49 +69,4 @@ describe("BlockArchiveRepository", function () {
     savedBlock2.message.slot = sampleBlock.message.slot;
     expect(ssz.phase0.SignedBeaconBlock.equals(savedBlock1, savedBlock2)).to.equal(true);
   });
-
-  it("batchPutBinary should be faster than batchPut", async () => {
```
Comment (dapplion) [inline packages/beacon-node/test/e2e/db/api/beacon/repositories/blockArchive.test.ts:74]: This is not a test but a very old performance test with winston profiler. Removed as it's not even tracked

File: packages/cli/src/cmds/beacon/options.ts [hunk 3]
```diff
@@ -63,33 +63,49 @@ export const beaconExtraOptions: ICliCommandOptions<IBeaconExtraArgs> = {
...
+    choices: logFormats,
+    type: "string",
+  },
+
+  logLevelModule: {
+    hidden: true,
+    description: "Set log level for a specific module by name: 'chain=debug' or 'network=debug,chain=debug'",
+    type: "array",
+    string: true,
+    coerce: (args: string[]) => args.map((item) => item.split(",")).flat(1),
```
Comment (dapplion) [inline packages/cli/src/cmds/beacon/options.ts:108]: New option that allows to customize arbitrary modules without dedicated flags per module

File: packages/cli/src/util/logger.ts [hunk 4]
```diff
@@ -1,47 +1,88 @@
...
+    transports.push(
+      rotate
+        ? new DailyRotateFile({
+            level: args.logFileLevel,
+            //insert the date pattern in filename before the file extension.
+            filename: filename.replace(/\.(?=[^.]*$)|$/, "-%DATE%$&"),
+            datePattern: "YYYY-MM-DD",
+            handleExceptions: true,
+            maxFiles: args.logFileDailyRotate ?? defaultLogMaxFiles,
+            auditFile: path.join(path.dirname(filename), ".log_rotate_audit.json"),
```
Comment (dapplion) [inline packages/cli/src/util/logger.ts:78]: For dockerized setups the audit file was placed in the process root and lost on restart. So DailyRotateFile could not clean the log files.

File: packages/cli/src/util/loggerConsoleTransport.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  setModuleLevel(module: string, level: LogLevel): void {
+    this.levelByModule.set(module, level);
+  }
+
+  deleteModuleLevel(module: string): boolean {
+    return this.levelByModule.delete(module);
+  }
+
+  _write(info: LogInfo, enc: BufferEncoding, callback: (error?: Error | null | undefined) => void): void {
```
Comment (dapplion) [inline packages/cli/src/util/loggerConsoleTransport.ts:39]: A bit intrusive into Winston mechanics but this is the easiest solution to have custom module levels by far from the few options I've tried

File: packages/cli/test/unit/cmds/beacon.test.ts [hunk 6]
```diff
@@ -60,7 +60,7 @@ describe("cmds / beacon / args handler", () => {
     fs.writeFileSync(peerIdFile, JSON.stringify(prevPeerId.toJSON()));
 
     const {peerId} = await runBeaconHandlerInit({
-      peerIdFile,
+      // peerIdFile,
```
Comment (dapplion) [inline packages/cli/test/unit/cmds/beacon.test.ts:63]: This test is skipped, not sure why Typescript didn't complain about this one before

File: packages/utils/src/logger/format.ts [hunk 7]
```diff
@@ -85,7 +85,7 @@ function humanReadableTemplateFn(_info: {[key: string]: any; level: string; mess
 
   if (info.timestamp) str += info.timestamp;
 
-  str += `[${infoString.toUpperCase()}] ${info.level.padStart(infoPad)}: ${info.message}`;
+  str += `[${infoString}] ${info.level.padStart(infoPad)}: ${info.message}`;
```
Comment (dapplion) [inline packages/utils/src/logger/format.ts:88]: Do not mutate module field to keep it equal to what's there in the code

**Comment by dapplion** [comment]
Tested locally with dev command and works well :+1: 

```
lodestar$ ./packages/cli/bin/lodestar.js dev --startValidators 0..7 --logLevelModule chain=debug

....
Eph 0/1 3.002[]                 info: Synced - slot: 1 - head: 1 0x2144…5eb4 - finalized: 0x0000…0000:0 - peers: 0
Eph 0/1 4.135[]                 info: Published aggregateAndProofs slot=1, index=0, count=1
Eph 0/2 0.009[chain]         verbose: Clock slot slot=2
Eph 0/2 0.026[]                 warn: Proposer duties re-org. This may happen from time to time priorDependentRoot=0xbda398f7e6216544a3d42e8b26ef79d31022585485e0cd6dafca5f16cdf5cd6c, dependentRoot=0x730702253512af6fd7bbd286acc2fa0e32c9b13e8f588100a3d38e02cc7a4fc9
Eph 0/2 0.047[chain]         verbose: New chain head headSlot=2, headRoot=0xfaaa45b02594122096125e5e7ecc95d34691a518d79c5fc3eec843a456d112d2, delaySec=0.04699993133544922
```

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**